spotify

```diff
@@ -119,18 +149,57 @@ while [ $# -gt 0 ]; do
                 array=( $@ );
                 len=${#array[@]};
                 SPOTIFY_SEARCH_API="https://api.spotify.com/v1/search"
+                SPOTIFY_TOKEN_URI="https://accounts.spotify.com/api/token"
+                if [ -z "${CLIENT_ID}" ]; then
+                    cecho "You are missing a Client ID, please update ${USER_CONFIG_FILE}"
+                    showAPIHelp
+                    exit 1
+                fi
+                if [ -z "${CLIENT_SECRET}" ]; then
+                    cecho "You are missing a Client Secret, please update ${USER_CONFIG_FILE}"
+                    showAPIHelp
+                    exit 1
+                fi
+                SHPOTIFY_CREDENTIALS=$(printf "${CLIENT_ID}:${CLIENT_SECRET}" | base64 --wrap=0)
                 SPOTIFY_PLAY_URI="";
 
                 searchAndPlay() {
                     type="$1"
                     Q="$2"
 
+                    cecho "Authorizing with Spotify API"
+                    SPOTIFY_TOKEN_RESPONSE_DATA=$( \
+                        curl "${SPOTIFY_TOKEN_URI}" \
+                            --silent \
+                            -X "POST" \
+                            -H "Authorization: Basic ${SHPOTIFY_CREDENTIALS}" \
+                            -d "grant_type=client_credentials" \
+                    )
+                    if ! [[ "${SPOTIFY_TOKEN_RESPONSE_DATA}" =~ "access_token" ]]; then
+                        cecho "Autorization failed, please check ${USER_CONFG_FILE}"
+                        cecho "${SPOTIFY_TOKEN_RESPONSE_DATA}"
+                        showAPIHelp
+                        exit 1
+                    fi
+                    SPOTIFY_ACCESS_TOKEN=$( \
+                        printf "${SPOTIFY_TOKEN_RESPONSE_DATA}" \
+                        | grep -E -o '"access_token":".*",' \
+                        | sed 's/"access_token"://g' \
+                        | sed 's/"//g' \
+                        | sed 's/,.*//g' \
+                    )
+
                     cecho "Searching ${type}s for: $Q";
 
                     SPOTIFY_PLAY_URI=$( \
-                        curl -s -G $SPOTIFY_SEARCH_API --data-urlencode "q=$Q" -d "type=$type&limit=1&offset=0" -H "Accept: application/json" \
+                        curl -s -G $SPOTIFY_SEARCH_API \
+                            -H "Authorization: Bearer ${SPOTIFY_ACCESS_TOKEN}" \
+                            -H "Accept: application/json" \
+                            --data-urlencode "q=$Q" \
+                            -d "type=$type&limit=1&offset=0" \
                         | grep -E -o "spotify:$type:[a-zA-Z0-9]+" -m 1
```

Comment:
I can also add a grep/error message here if the fault is in the oauth when I get some more time to work on this